src/Core/src/Handlers/Image/ImageHandler.iOS.cs

```diff
@@ -38,11 +35,16 @@ public static void MapBackground(IImageHandler handler, IImage image)
 		public static void MapSource(IImageHandler handler, IImage image) =>
 			MapSourceAsync(handler, image).FireAndForget(handler);
 
-		public static Task MapSourceAsync(IImageHandler handler, IImage image) =>
-			handler.SourceLoader.UpdateImageSourceAsync();
+		public static async Task MapSourceAsync(IImageHandler handler, IImage image) =>
+			await handler.SourceLoader.UpdateImageSourceAsync();
+		
+		void IImageSourcePartSetter.SetImageSource(UIImage? source)
+		{
+			PlatformView.Image = source;
 
-		void IImageSourcePartSetter.SetImageSource(UIImage? obj) =>
-			PlatformView.Image = obj;
+			if (VirtualView.Source is IStreamImageSource)
```

Comment:
Do we need the check? Shouldn't we always invalidate?

Reply:
Tested with the rest of types and is only necessary with Streams.